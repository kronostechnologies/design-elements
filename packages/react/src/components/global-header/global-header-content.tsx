import { FunctionComponent, PropsWithChildren, ReactNode, useState } from 'react';
import styled, { useTheme } from 'styled-components';
import { focus } from '../../utils/css-state';
import { useDeviceContext } from '../device-context-provider/device-context-provider';
import { Icon } from '../icon/icon';
import { SideDrawer } from '../side-drawer/side-drawer';

const BurgerButton = styled.button`
    appearance: none;
    background-color: transparent;
    border: 1px solid transparent;
    margin: 0;
    padding: 0;

    ${focus};
`;

const Container = styled.div`
    align-items: center;
    color: ${(props) => props.theme.component['global-header-content-text-color']};
    display: flex;
    justify-content: flex-end;
    width: 100%;

    > * + * {
        margin-left: var(--spacing-1x);
    }
`;

interface Props {
    mobileDrawerContent: ReactNode;
}

export const Content: FunctionComponent<PropsWithChildren<Props>> = ({ children, mobileDrawerContent }) => {
    const { isMobile } = useDeviceContext();
    const [drawerOpen, setDrawerOpen] = useState(false);
    const theme = useTheme();

    if (isMobile && mobileDrawerContent) {
        return (
            <>
                <BurgerButton
                    aria-expanded={drawerOpen}
                    aria-controls="menu-drawer"
                    aria-label="Navigation"
                    type="button"
                    onClick={() => setDrawerOpen(!drawerOpen)}
                >
                    <Icon name="menu" color={theme.greys.white} />
                </BurgerButton>
                <SideDrawer id="menu-drawer" open={drawerOpen}>
                    {mobileDrawerContent}
                </SideDrawer>
            </>
        );
    }

    return (
        <Container>
            {children}
        </Container>
    );
};
