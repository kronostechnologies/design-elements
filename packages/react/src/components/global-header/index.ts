export { GlobalHeader } from './global-header';
export type { GlobalHeaderProps } from './global-header';
export type { LogoName } from './logo';
