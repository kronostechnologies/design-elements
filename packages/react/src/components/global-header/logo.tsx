import { ReactElement } from 'react';

import EquisoftAnalyzeMobile from '../../logos/analyze/analyze-reverse-mobile.svg';
import EquisoftAnalyze from '../../logos/analyze/analyze-reverse.svg';
import EquisoftApplyMobile from '../../logos/apply/apply-reverse-mobile.svg';
import EquisoftApply from '../../logos/apply/apply-reverse.svg';
import EquisoftCentralizeMobile from '../../logos/centralize/centralize-reverse-mobile.svg';
import EquisoftCentralize from '../../logos/centralize/centralize-reverse.svg';
import EquisoftConnect from '../../logos/connect/connect-reverse.svg';
import EquisoftDesignMobile from '../../logos/design/design-reverse-mobile.svg';
import EquisoftDesign from '../../logos/design/design-reverse.svg';
import EquisoftIllustrateMobile from '../../logos/illustrate/illustrate-reverse-mobile.svg';
import EquisoftIllustrate from '../../logos/illustrate/illustrate-reverse.svg';
import EquisoftDefault from '../../logos/logo-equisoft-reversed.svg';
import EquisoftManageMobile from '../../logos/manage/manage-reverse-mobile.svg';
import EquisoftManage from '../../logos/manage/manage-reverse.svg';
import EquisoftPlan from '../../logos/plan/plan-reverse.svg';

const logoMapping = {
    default: {
        desktop: EquisoftDefault,
        mobile: EquisoftDefault,
    },
    analyze: {
        desktop: EquisoftAnalyze,
        mobile: EquisoftAnalyzeMobile,
    },
    apply: {
        desktop: EquisoftApply,
        mobile: EquisoftApplyMobile,
    },
    centralize: {
        desktop: EquisoftCentralize,
        mobile: EquisoftCentralizeMobile,
    },
    connect: {
        desktop: EquisoftConnect,
        mobile: EquisoftConnect,
    },
    design: {
        desktop: EquisoftDesign,
        mobile: EquisoftDesignMobile,
    },
    illustrate: {
        desktop: EquisoftIllustrate,
        mobile: EquisoftIllustrateMobile,
    },
    manage: {
        desktop: EquisoftManage,
        mobile: EquisoftManageMobile,
    },
    plan: {
        desktop: EquisoftPlan,
        mobile: EquisoftPlan,
    },
};

export type LogoName = keyof typeof logoMapping;

interface LogoProps {
    name?: LogoName;
    mobile?: boolean;
}

const smallLogos: LogoName[] = ['default', 'plan', 'connect'];

function getLogoHeight({ name, mobile }: Required<LogoProps>): string {
    if (!mobile && smallLogos.includes(name)) {
        return '24px';
    }
    return '100%';
}

export const Logo = ({ name = 'default', mobile = false }: LogoProps): ReactElement | null => {
    const Component = logoMapping[name][mobile ? 'mobile' : 'desktop'];

    const height = getLogoHeight({ name, mobile });

    return <Component style={{ height }} focusable="false" aria-hidden="true" />;
};
