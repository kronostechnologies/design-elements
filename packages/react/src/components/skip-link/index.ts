export { SkipLink } from './skip-link';
export type { SkipLinkProps } from './skip-link';
