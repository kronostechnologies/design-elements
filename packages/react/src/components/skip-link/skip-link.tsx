import { MouseEvent, VoidFunctionComponent } from 'react';
import styled from 'styled-components';
import { useTranslation } from '../../i18n/use-translation';
import { defaultButtonStyles } from '../buttons/abstract-button';
import { useDeviceContext } from '../device-context-provider/device-context-provider';

const StyledLink = styled.a<{ isMobile: boolean }>`
    ${defaultButtonStyles}

    color: ${({ theme }) => theme.colors['primary-1.1']};
    font-size: ${({ isMobile }) => (isMobile ? 1 : 0.875)}rem;
    font-weight: var(--font-normal);
    letter-spacing: 0.015rem;
    overflow: hidden;
    position: absolute;
    text-transform: unset;
    white-space: nowrap;

    &:not(:focus) {
        clip: rect(1px, 1px, 1px, 1px);
        height: 1px;
        width: 1px;
    }

    &:focus {
        background-color: ${({ theme }) => theme.colors.white};
        position: absolute;
    }
`;

export interface SkipLinkProps {
    className?: string;
    href: string;
    onClick?(event: MouseEvent<HTMLAnchorElement>): void;
}

export const SkipLink: VoidFunctionComponent<SkipLinkProps> = ({
    className, href, onClick,
}) => {
    const { isMobile } = useDeviceContext();
    const { t } = useTranslation('skip-link');

    return (
        <StyledLink
            className={className}
            href={href}
            isMobile={isMobile}
            onClick={onClick}
        >
            {t('label')}
        </StyledLink>
    );
};
