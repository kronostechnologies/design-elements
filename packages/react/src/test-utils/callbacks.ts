export function doNothing(): void {
    // do nothing
}
