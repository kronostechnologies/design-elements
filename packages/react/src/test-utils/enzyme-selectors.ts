import { ReactWrapper, ShallowWrapper } from 'enzyme';

enum ModifierEnum {
    StartsWith = '^',
    EndsWith = '$',
    ContainsOptionalHyphen = '|',
    Contains = '*',
    ContainsWord = '~',
    None = '',
}
type SelectorModifier = `${ModifierEnum}`;

function filterHtmlNodesOnly<W extends ReactWrapper | ShallowWrapper>(
    nodes: W,
): W {
    return nodes
        .hostNodes()
        .filterWhere((x) => x.name() !== null && !x.name().startsWith('styled.')) as W;
}

export function findByTestId<W extends ReactWrapper | ShallowWrapper>(
    wrapper: W,
    testId: string,
    options?: {
        modifier?: SelectorModifier,
        nodeName?: string,
        // Don't use with shallow wrapper as it contains styled node over the html nodes.
        htmlNodesOnly?: boolean,
    },
): W {
    const modifier = options?.modifier ?? '';

    let nodes = wrapper.find(`[data-testid${modifier}="${testId}"]`) as W;

    if (options?.nodeName) {
        nodes = nodes.filterWhere((x) => x.name() === options.nodeName) as W;
    }
    if (options?.htmlNodesOnly) {
        nodes = filterHtmlNodesOnly(nodes);
    }

    return nodes;
}

export function getByTestId<W extends ReactWrapper | ShallowWrapper>(
    wrapper: W,
    testId: string,
    options?: {
        modifier?: SelectorModifier,
        nodeName?: string,
        // Don't use with shallow wrapper as it contains styled node instead of the html nodes.
        htmlNodesOnly?: boolean,
    },
): W {
    return findByTestId(wrapper, testId, options).at(0) as W;
}
