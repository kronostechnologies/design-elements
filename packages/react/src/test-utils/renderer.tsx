import { ThemeWrapped } from '@design-elements/test-utils/theme-wrapped';
import { mount, ReactWrapper, render } from 'enzyme';
import React, { Component, ReactElement } from 'react';
import { MemoryRouter } from 'react-router-dom';
import { DeviceContextProvider } from '../components/device-context-provider/device-context-provider';

export function mountWithProviders<C extends Component, P = C['props'], S = C['state']>(
    component: ReactElement<P>,
): ReactWrapper<P, S, C> {
    return mount(component, {
        wrappingComponent: AllProviders,
    });
}

export function renderWithProviders(
    component: ReactElement,
): Cheerio {
    return render(<AllProviders>{component}</AllProviders>);
}

function AllProviders({ children }: { children: ReactElement }): ReactElement {
    return (
        <MemoryRouter>
            <DeviceContextProvider>
                {ThemeWrapped(children)}
            </DeviceContextProvider>
        </MemoryRouter>
    );
}
