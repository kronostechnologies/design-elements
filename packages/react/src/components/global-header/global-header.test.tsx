import { shallow } from 'enzyme';
import { getByTestId } from '../../test-utils/enzyme-selectors';
import { mountWithProviders, renderWithProviders } from '../../test-utils/renderer';
import { SkipLinkProps } from '../skip-link/skip-link';
import { GlobalHeader } from './global-header';

describe('Global Header', () => {
    it('Matches the snapshot (desktop)', () => {
        const { container } = renderWithProviders(
            <GlobalHeader mobileDrawerContent={(<p>Test</p>)}>
                Hello, World!
            </GlobalHeader>,
        );

        expect(container.firstChild).toMatchSnapshot();
    });

    it('Matches the snapshot (mobile)', () => {
        const { container } = renderWithProviders(
            <GlobalHeader mobileDrawerContent={(<p>Test</p>)}>
                Hello, World!
            </GlobalHeader>,
            'mobile',
        );

        expect(container.firstChild).toMatchSnapshot();
    });

    it('mobileDrawerContent prop adds a side drawer and burger button in mobile', () => {
        const { container } = renderWithProviders(
            <GlobalHeader mobileDrawerContent={(<p>Test</p>)}>
                Hello, World!
            </GlobalHeader>,
            'mobile',
        );

        expect(container.firstChild).toMatchSnapshot();
    });

    describe('SkipLink', () => {
        it('should not exist when skipLink is not defined', () => {
            const wrapper = shallow(
                <GlobalHeader>
                    Test
                </GlobalHeader>,
            );

            expect(getByTestId(wrapper, 'skip-link').exists()).toBe(false);
        });

        it('should receive props when skipLink is defined', () => {
            const skipLink: SkipLinkProps = {
                href: '#test',
                onClick: jest.fn(),
            };

            const wrapper = shallow(
                <GlobalHeader skipLink={skipLink}>
                    Test
                </GlobalHeader>,
            );

            const skipLinkRef = getByTestId(wrapper, 'skip-link');
            expect(skipLinkRef.prop('href')).toBe(skipLink.href);
            expect(skipLinkRef.prop('onClick')).toBe(skipLink.onClick);
        });
    });

    describe('logo', () => {
        it('should use react-router-link when usesReactRouter is true', () => {
            const wrapper = shallow(<GlobalHeader usesReactRouter>test</GlobalHeader>);

            expect(getByTestId(wrapper, 'logo-react-router-link').exists()).toBe(true);
        });

        it('should use html-link when usesReactRouter is false', () => {
            const wrapper = shallow(<GlobalHeader usesReactRouter={false}>test</GlobalHeader>);

            expect(getByTestId(wrapper, 'logo-html-link').exists()).toBe(true);
        });

        it('contains app-title when appTitleDesktop prop is set given device is desktop', () => {
            const wrapper = mountWithProviders(
                <GlobalHeader appTitleDesktop="test">test</GlobalHeader>,
                { wrappingComponentProps: { staticDevice: 'desktop' } },
            );

            expect(getByTestId(wrapper, 'app-title').exists()).toBe(true);
        });

        it('does not contain app-title when device is mobile', () => {
            const wrapper = mountWithProviders(
                <GlobalHeader appTitleDesktop="test">test</GlobalHeader>,
                { wrappingComponentProps: { staticDevice: 'mobile' } },
            );

            expect(getByTestId(wrapper, 'app-title').exists()).toBe(false);
        });
    });
});
