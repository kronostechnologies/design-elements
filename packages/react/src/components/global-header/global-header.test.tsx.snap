// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Global Header Matches the snapshot (desktop) 1`] = `
.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  color: #FFFFFF;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  width: 100%;
}

.c2 > * + * {
  margin-left: var(--spacing-1x);
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #012639;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: var(--size-3x);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: var(--spacing-1x) var(--spacing-2x);
  position: relative;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.5rem;
  font-weight: var(--font-bold);
  height: 100%;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c1:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1 > * {
  height: 100%;
}

<header
  class="c0"
>
  <a
    aria-label="Home"
    class="c1"
    data-testid="logo-react-router-link"
    href="/"
  >
    <svg
      aria-hidden="true"
      focusable="false"
      style="height:100%"
    />
  </a>
  <div
    class="c2"
  >
    Hello, World!
  </div>
</header>
`;

exports[`Global Header Matches the snapshot (mobile) 1`] = `
.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.5rem;
  box-sizing: border-box;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.875rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.033125rem;
  -moz-letter-spacing: 0.033125rem;
  -ms-letter-spacing: 0.033125rem;
  letter-spacing: 0.033125rem;
  line-height: 1.5rem;
  min-height: var(--size-3x);
  min-width: 2rem;
  outline: none;
  padding: 0 var(--spacing-3x);
  text-transform: uppercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c2:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c2 > svg {
  color: inherit;
  height: var(--size-1halfx);
  width: var(--size-1halfx);
}

.c3 {
  background-color: transparent;
  border-color: transparent;
  color: #FFFFFF;
  padding: 0;
  width: var(--size-3x);
}

.c3 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c3:focus {
  box-shadow: 0 0 0 2px #84C6EA;
  outline: 2px solid #006296;
  outline-offset: -2px;
}

.c3:hover,
.c3[aria-expanded='true'] {
  background-color: rgb(0 0 0 / 0.5);
  border-color: transparent;
  color: #FFFFFF;
}

.c3:disabled {
  background-color: transparent;
  border-color: transparent;
  color: #006296;
}

.c3 > svg {
  height: var(--size-1halfx);
  width: var(--size-1halfx);
}

.c6 {
  background-color: #FFFFFF;
  box-shadow: 0 6px 10px 0 rgb(0 0 0 / 0.1);
  height: calc(100vh - 56px);
  right: 0;
  overflow-x: hidden;
  overflow-y: auto;
  position: fixed;
  top: 56px;
  -webkit-transform: translate(100%);
  -ms-transform: translate(100%);
  transform: translate(100%);
  -webkit-transition: -webkit-transform 300ms;
  -webkit-transition: transform 300ms;
  transition: transform 300ms;
  width: 100%;
  z-index: 100;
}

.c5 > .c5 {
  height: 100%;
  top: 0;
}

.c4 {
  margin: 0;
  padding: 0;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #012639;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: var(--size-3halfx);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: var(--spacing-2x);
  position: relative;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.5rem;
  font-weight: var(--font-bold);
  height: 100%;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c1:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1 > * {
  height: 100%;
}

<header
  class="c0"
>
  <a
    aria-label="Home"
    class="c1"
    data-testid="logo-react-router-link"
    href="/"
  >
    <svg
      aria-hidden="true"
      focusable="false"
      style="height:100%"
    />
  </a>
  <button
    aria-controls="menu-drawer"
    aria-expanded="false"
    aria-label="Navigation"
    class="c2 c3 c4"
    type="button"
  >
    <svg
      aria-hidden="true"
      color="currentColor"
      focusable="false"
      height="24"
      width="24"
    />
  </button>
  <div
    aria-hidden="true"
    class="c5 c6"
    id="menu-drawer"
    origin="right"
    width="100%"
  >
    <p>
      Test
    </p>
  </div>
</header>
`;

exports[`Global Header mobileDrawerContent prop adds a side drawer and burger button in mobile 1`] = `
.c2 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
  background: inherit;
  border: 1px solid;
  border-radius: 1.5rem;
  box-sizing: border-box;
  color: inherit;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  font-family: inherit;
  font-size: 0.875rem;
  font-weight: var(--font-bold);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-letter-spacing: 0.033125rem;
  -moz-letter-spacing: 0.033125rem;
  -ms-letter-spacing: 0.033125rem;
  letter-spacing: 0.033125rem;
  line-height: 1.5rem;
  min-height: var(--size-3x);
  min-width: 2rem;
  outline: none;
  padding: 0 var(--spacing-3x);
  text-transform: uppercase;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
}

.c2 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c2:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c2 > svg {
  color: inherit;
  height: var(--size-1halfx);
  width: var(--size-1halfx);
}

.c3 {
  background-color: transparent;
  border-color: transparent;
  color: #FFFFFF;
  padding: 0;
  width: var(--size-3x);
}

.c3 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c3:focus {
  box-shadow: 0 0 0 2px #84C6EA;
  outline: 2px solid #006296;
  outline-offset: -2px;
}

.c3:hover,
.c3[aria-expanded='true'] {
  background-color: rgb(0 0 0 / 0.5);
  border-color: transparent;
  color: #FFFFFF;
}

.c3:disabled {
  background-color: transparent;
  border-color: transparent;
  color: #006296;
}

.c3 > svg {
  height: var(--size-1halfx);
  width: var(--size-1halfx);
}

.c6 {
  background-color: #FFFFFF;
  box-shadow: 0 6px 10px 0 rgb(0 0 0 / 0.1);
  height: calc(100vh - 56px);
  right: 0;
  overflow-x: hidden;
  overflow-y: auto;
  position: fixed;
  top: 56px;
  -webkit-transform: translate(100%);
  -ms-transform: translate(100%);
  transform: translate(100%);
  -webkit-transition: -webkit-transform 300ms;
  -webkit-transition: transform 300ms;
  transition: transform 300ms;
  width: 100%;
  z-index: 100;
}

.c5 > .c5 {
  height: 100%;
  top: 0;
}

.c4 {
  margin: 0;
  padding: 0;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background: #012639;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: var(--size-3halfx);
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: var(--spacing-2x);
  position: relative;
}

.c1 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  font-size: 1.5rem;
  font-weight: var(--font-bold);
  height: 100%;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c1 {
  outline: 2px solid transparent;
  outline-offset: -2px;
}

.c1:focus {
  box-shadow: 0 0 0 2px #006296;
  outline: 2px solid #84C6EA;
  outline-offset: -2px;
}

.c1 > * {
  height: 100%;
}

<header
  class="c0"
>
  <a
    aria-label="Home"
    class="c1"
    data-testid="logo-react-router-link"
    href="/"
  >
    <svg
      aria-hidden="true"
      focusable="false"
      style="height:100%"
    />
  </a>
  <button
    aria-controls="menu-drawer"
    aria-expanded="false"
    aria-label="Navigation"
    class="c2 c3 c4"
    type="button"
  >
    <svg
      aria-hidden="true"
      color="currentColor"
      focusable="false"
      height="24"
      width="24"
    />
  </button>
  <div
    aria-hidden="true"
    class="c5 c6"
    id="menu-drawer"
    origin="right"
    width="100%"
  >
    <p>
      Test
    </p>
  </div>
</header>
`;
